mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -963,6 +963,15 @@ impl TakerSwap {
             )]));
         }
 
+        if !self
+            .maker_coin
+            .validate_secret_hash(maker_data.secret_hash(), self.r().secret.0.as_slice())
```

Comment:
As I suggested above, we can simply check `if maker_data.secret_hash() == self.r().secret.0`

Reply:
done